code/modules/industrial_lift/tram_controls.dm

```diff
@@ -172,6 +187,53 @@
 
 	computer.try_send_tram(destination)
 
+/obj/machinery/computer/tram_controls/proc/set_controls_display()
+	var/datum/lift_master/tram/tram = tram_ref?.resolve()
```

Comment:
you don't need to use tram_ref here, the source argument of set_controls_display will be the source of the signal that called it, IE: the tram
also use SIGNAL_HANDLER here, it prevents sleeping in this proc to ensure signals work proper

Reply:
Yep, fixed as per your direction.